### PR TITLE
Issue 18318 limit access to app

### PR DESCRIPTION
**Pull request by wezell**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java [hunk 1]
```diff
@@ -157,5 +158,36 @@ public void test_unable_to_modify_userid_in_velocity_session() throws Exception{
...
+
+        final HttpServletResponse response = new MockHttpResponse().response();
+
+
+        ChainedContext context = (ChainedContext) VelocityUtil.getWebContext(request, response);
+
+        assert (context.getServletContext() == null);
+        assert (context.get("application") == null);
+        assert (context.getRequest().getServletContext() == null);
+        assert (context.getAttribute("application") == null);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:185]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java [hunk 2]
```diff
@@ -112,5 +112,35 @@ public void setMaxInactiveInterval(int arg0) {
...
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((session == null) ? 0 : session.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java:126]: Codacy Issue found: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java [hunk 3]
```diff
@@ -157,5 +158,36 @@ public void test_unable_to_modify_userid_in_velocity_session() throws Exception{
...
+        final HttpServletRequest request = new MockSessionRequest(
+                        new MockAttributeRequest(new MockHttpRequest("localhost", "/").request()).request()).request();
+
+        final HttpServletResponse response = new MockHttpResponse().response();
+
+
+        ChainedContext context = (ChainedContext) VelocityUtil.getWebContext(request, response);
+
+        assert (context.getServletContext() == null);
+        assert (context.get("application") == null);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:183]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java [hunk 4]
```diff
@@ -87,31 +86,34 @@ public ChainedContext(Context ctx,
...
-                          HttpServletResponse response,
-                          ServletContext application)
+                          HttpServletResponse response)
     {
-        this(null, velocity, request, response, application);
+        this(null, velocity, request, response );
+    }
+    public ChainedContext(VelocityEngine velocity,
+                    HttpServletRequest request,
+                    HttpServletResponse response,ServletContext application)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java:101]: Codacy Issue found: [Parameter 'response' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

Comment (dev-dotcms) [inline dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java:101]: Codacy Issue found: [Avoid unused constructor parameters such as 'application'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java [hunk 5]
```diff
@@ -87,31 +86,34 @@ public ChainedContext(Context ctx,
...
                           HttpServletRequest request,
-                          HttpServletResponse response,
-                          ServletContext application)
+                          HttpServletResponse response)
     {
         super(null, ctx);
 
         this.velocity = velocity;
-        this.request = request;
+        this.request = (request instanceof VelocityRequestWrapper) ? request : new VelocityRequestWrapper(request);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java:113]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java [hunk 6]
```diff
@@ -112,5 +112,35 @@ public void setMaxInactiveInterval(int arg0) {
...
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        VelocitySessionWrapper other = (VelocitySessionWrapper) obj;
+        if (session == null) {
+            if (other.session != null)
+                return false;
+        } else if (!session.equals(other.session))
+            return false;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java:136]: Codacy Issue found: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java [hunk 7]
```diff
@@ -157,5 +158,36 @@ public void test_unable_to_modify_userid_in_velocity_session() throws Exception{
...
+
+        final HttpServletRequest request = new MockSessionRequest(
+                        new MockAttributeRequest(new MockHttpRequest("localhost", "/").request()).request()).request();
+
+        final HttpServletResponse response = new MockHttpResponse().response();
+
+
+        ChainedContext context = (ChainedContext) VelocityUtil.getWebContext(request, response);
+
+        assert (context.getServletContext() == null);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:182]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java [hunk 8]
```diff
@@ -112,5 +112,35 @@ public void setMaxInactiveInterval(int arg0) {
...
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((session == null) ? 0 : session.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java:124]: Codacy Issue found: [Parameter 'obj' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java [hunk 9]
```diff
@@ -157,5 +158,36 @@ public void test_unable_to_modify_userid_in_velocity_session() throws Exception{
...
+        assert (context.get("application") == null);
+        assert (context.getRequest().getServletContext() == null);
+        assert (context.getAttribute("application") == null);
+        assert (context.getVelocityContext().get("application") == null);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:186]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java [hunk 10]
```diff
@@ -157,5 +158,36 @@ public void test_unable_to_modify_userid_in_velocity_session() throws Exception{
...
+                        new MockAttributeRequest(new MockHttpRequest("localhost", "/").request()).request()).request();
+
+        final HttpServletResponse response = new MockHttpResponse().response();
+
+
+        ChainedContext context = (ChainedContext) VelocityUtil.getWebContext(request, response);
+
+        assert (context.getServletContext() == null);
+        assert (context.get("application") == null);
+        assert (context.getRequest().getServletContext() == null);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:184]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java [hunk 11]
```diff
@@ -157,5 +158,36 @@ public void test_unable_to_modify_userid_in_velocity_session() throws Exception{
...
+    @Test
+    public void test_servlet_context_is_not_available_in_velocity() throws Exception {
+
+        final HttpServletRequest request = new MockSessionRequest(
+                        new MockAttributeRequest(new MockHttpRequest("localhost", "/").request()).request()).request();
+
+        final HttpServletResponse response = new MockHttpResponse().response();
+
+
+        ChainedContext context = (ChainedContext) VelocityUtil.getWebContext(request, response);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:180]: Codacy Issue found: [Local variable 'context' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java [hunk 12]
```diff
@@ -157,5 +158,36 @@ public void test_unable_to_modify_userid_in_velocity_session() throws Exception{
...
+     * https://github.com/dotCMS/core/issues/18318
+     * 
+     * The servlet context is too dangerous to expose to velocity and allows for arbitrary 
+     * code execution
+     * 
+     * @throws Exception
+     */
+    
+    @Test
+    public void test_servlet_context_is_not_available_in_velocity() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:172]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java [hunk 13]
```diff
@@ -87,31 +86,34 @@ public ChainedContext(Context ctx,
...
+    {
+        this(null, velocity, request, response );
     }
-    
     public ChainedContext(Context ctx,
                           VelocityEngine velocity,
                           HttpServletRequest request,
-                          HttpServletResponse response,
-                          ServletContext application)
+                          HttpServletResponse response)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java:108]: Codacy Issue found: [Parameter 'response' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java [hunk 14]
```diff
@@ -112,5 +112,35 @@ public void setMaxInactiveInterval(int arg0) {
...
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        VelocitySessionWrapper other = (VelocitySessionWrapper) obj;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java:131]: Codacy Issue found: [Local variable 'other' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

**Review by jdotcms — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:180]: Codacy Issue found: [Local variable 'context' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:184]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java:113]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java:101]: Codacy Issue found: [Parameter 'response' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:185]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java:124]: Codacy Issue found: [Parameter 'obj' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java (hunk 14 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java:131]: Codacy Issue found: [Local variable 'other' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:182]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocitySessionWrapper.java:126]: Codacy Issue found: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:186]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:172]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/org/apache/velocity/tools/view/context/ChainedContext.java:101]: Codacy Issue found: [Avoid unused constructor parameters such as 'application'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java:183]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)

**Pull request merged**